src/main/java/view/ClubDescriptionExploreContainer.java

```diff
@@ -0,0 +1,22 @@
+package view;
+
+import javax.swing.BoxLayout;
+import javax.swing.JPanel;
+
+import entity.user.Club;
+import interface_adapter.student_logged_in.explore_clubs.ExploreClubsPresenter;
+import interface_adapter.student_logged_in.explore_clubs.ExploreClubsState;
+
+/**
+ * Container for all explore page club mini panels.
+ */
+public class ClubDescriptionExploreContainer extends JPanel {
+    public ClubDescriptionExploreContainer(ExploreClubsState exploreClubsState,
+                                           ExploreClubsPresenter exploreClubsPresenter) {
+        this.setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+
+        for (Club club : exploreClubsState.getClubs().getAll()) {
+            this.add(new ClubMiniPanel(club, exploreClubsPresenter));
```

Comment:
looks good but feels wrong to have clubs in views. I don't see any problem with CA though, since this view is dedicated to only clubs, and thus there would be no problem with SOLID

Reply:
Having Clubs in the view does actually violate CA, since Club is an entity. This needs to be redone by having the use case interactor read the clubs you want and having the output data contain the data you read without any entities.

Reply:
I see. Will do!